### PR TITLE
docs: support notImplementedHide

### PR DESCRIPTION
**Pull request by zirain**
Fixes: https://github.com/envoyproxy/gateway/issues/2575

### CODE REVIEW

**Comment by arkodg** [comment]
thanks for implementing this @zirain !
ptal @envoyproxy/gateway-maintainers , we need to make a decision here on, should we fork `elastic/crd-ref-docs` to implement this feature or temporarily circumvent this by just adding a doc string that says `unimplemented`

**Comment by zirain** [comment]
xref: https://github.com/elastic/crd-ref-docs/issues/70

**Comment by Xunzhuo** [comment]
Can we push this into upstream before landing it ?

**Comment by zirain** [comment]
wait for https://github.com/elastic/crd-ref-docs/pull/76

File: api/v1alpha1/backendtrafficpolicy_types.go
Comment (arkodg) [inline api/v1alpha1/backendtrafficpolicy_types.go:98]: like the UX !

File: tools/src/crd-ref-docs/go.mod
Comment (zirain) [inline tools/src/crd-ref-docs/go.mod:5]: update to latest

**Comment by arkodg** [comment]
🎉 thanks for driving this ! and making upstream changes as well !
some nits on naming, should we call this
* hidefromdoc
* hidefromdocs
* hidden
* not-implemented
* <other>

cc @envoyproxy/gateway-maintainers

**Comment by Xunzhuo** [comment]
Unimplemented

**Comment by zhaohuabing** [comment]
We can follow Envoy to use `not-implemented-hide`.

**Comment by zirain** [comment]
> Envoy uses a `not-implemented-hide` annotation in proto APIs for unimplemented features. How about using the same in EG?

I'm OK with this, it's clear enough to anyone who read it.

**Comment by AliceProxy** [comment]
> not-implemented-hide

Plus one for this

**Comment by guydc** [comment]
> not-implemented-hide

Also plus one for this

**Comment by arkodg** [comment]
the `not-implemented-hide`s have it !

**Comment by zirain** [comment]
> not-implemented-hide

cannot use `-`, rename to `notimplementedhide`

**Comment by arkodg** [comment]
`notimplementedhide` is really hard to read imo

**Comment by zirain** [comment]
> `notimplementedhide` is really hard to read imo

`NotImplementedHide` is better, it's case-sensitive.

**Comment by arkodg** [comment]
Thoughts on camelCase ? 'notImplementedHide' ?

**Comment by zirain** [comment]
> thoughts on camelCase ? `notImplementedHide` ?

sg

**Review by arkodg — APPROVED** [review]
🚀

**Review by zhaohuabing — APPROVED** [review]

**Pull request merged**
